devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -478,15 +478,19 @@ public void close() throws IOException {
                     }
                     if (!changed.isEmpty()) {
                         getLog().info("Changes detected to " + changed + ", restarting dev mode");
+
+                        // stop the runner before we build the new one as the debug port being free
+                        // is tested when building the runner
+                        runner.stop();
```

Comment:
Do we need to do this in Gradle as well?

Reply:
Your guess is as good as mine :). I will try to have a look.

Reply:
I don't see the restart logic in Gradle but I'm not very familiar with how it works.

Maybe @glefloch or @aloubyansky know more.

But I wouldn't block this PR as the current Maven behavior is a bit annoying.